### PR TITLE
Test cases for Vec

### PR DESCRIPTION
**Pull request by justkawal**
## Any follow ups?
Please review and approve PR: #62, before reviewing this PR.

## Description
* It adds test-cases for `Vec`.
* Combined Test-Cases:
    * `Vec<u8>`
    * `Vec<Vec<u8>>`
    * `Vec<u256>`
    * `Vec<i256>`
    * `Vec<(u32, u32, u16)>`
    * `Vec<(u32, Option<bool>)>`
    * `Vec<(String, bool)>`
    * `Vec<bool>`
    * `Vec<Option<bool>>` 
    * `Vec<Option<u8>>`
    * `Vec<Text>`

## How?
Testing multiple combination of test-cases helps to narrow down the breaking scenarios.

## Testing Plan 
Test the file `vec_test.dart`.

### CODE REVIEW

File: packages/polkadart_scale_codec/test/test_types/vec_test.dart [hunk 1]
```diff
@@ -0,0 +1,442 @@
...
+            codec.encodeToHex(registry.getIndex('Vec<u8>'), [255, 255]);
+        expect(expectedValue, encodedValue);
+      });
+
+      // Decode
+      test('When 0x08ffff is decoded then it should produce result: [255, 255]',
+          () {
+        final expectedValue = [255, 255];
+        final decodedValue =
+            codec.decodeBinary(registry.getIndex('Vec<u8>'), '0x08ffff');
```
Comment (gabrielokura) [inline packages/polkadart_scale_codec/test/test_types/vec_test.dart:50]: Is this name right? Aren't we decoding a hexadecimal string?

File: packages/polkadart_scale_codec/lib/src/core/codec.dart [hunk 2]
```diff
@@ -351,7 +351,12 @@ class Codec {
       assert(val == null);
```
Comment (gabrielokura) [inline packages/polkadart_scale_codec/lib/src/core/codec.dart:351]: this scenario isn't being reach in the tests.

**Review by gabrielokura — CHANGES_REQUESTED** [review]

File: packages/polkadart_scale_codec/test/test_types/vec_test.dart (hunk 1 above)
Comment (justkawal) [reply packages/polkadart_scale_codec/test/test_types/vec_test.dart:50]: Yes, it's right if you go through the code thoroughly then you'll find out that we encodes the value to hex-decimal and then decodes it back to value.

File: packages/polkadart_scale_codec/lib/src/core/codec.dart (hunk 2 above)
Comment (justkawal) [reply packages/polkadart_scale_codec/lib/src/core/codec.dart:351]: Brother, it's tuple test cases. The main motive is to test Vector and not Tuples.

File: packages/polkadart_scale_codec/test/test_types/vec_test.dart [hunk 3]
```diff
@@ -0,0 +1,442 @@
...
+  {
+    group('Test Vec<u8>:', () {
+      // Encode
+      test(
+          'When [255, 255] is encoded then it should produce result: \'0x08ffff\'',
+          () {
+        final String expectedValue = '0x08ffff';
+        final String encodedValue =
+            codec.encodeToHex(registry.getIndex('Vec<u8>'), [255, 255]);
+        expect(expectedValue, encodedValue);
```
[the remark this answers is not in the document]
Comment (justkawal) [reply packages/polkadart_scale_codec/test/test_types/vec_test.dart]: Done

File: packages/polkadart_scale_codec/test/test_types/vec_test.dart (hunk 1 above)
Comment (gabrielokura) [reply packages/polkadart_scale_codec/test/test_types/vec_test.dart:50]: `decodeBinary` means we are decoding a binary value, no? If we are decoding `encoded hexa-decimal` values shouldn't the function name be `decodeHex` or `decodeFromHex`?

File: packages/polkadart_scale_codec/lib/src/core/codec.dart [hunk 4]
```diff
@@ -351,7 +351,12 @@ class Codec {
...
       return;
     }
-    assertionCheck(val is List && def.tuple.length == val.length);
+    if (val is! List) {
+      throw UnexpectedCaseException(
```
[the remark this answers is not in the document]
Comment (justkawal) [reply packages/polkadart_scale_codec/lib/src/core/codec.dart]: Done

File: packages/polkadart_scale_codec/test/test_types/vec_test.dart (hunk 1 above)
Comment (justkawal) [reply packages/polkadart_scale_codec/test/test_types/vec_test.dart:50]: Changed to decodeHex. @gabrielokura 

Please consider to change this in your PR too parallely.

Comment (justkawal) [reply packages/polkadart_scale_codec/test/test_types/vec_test.dart:50]: Changed: 
`encodeToHex` to `encode`
`decodeBinary` to `decode`

**Review by leonardocustodio — APPROVED** [review]

**Review by gabrielokura — APPROVED** [review]

**Review by Lohann — APPROVED** [review]

**Pull request merged**